### PR TITLE
Simplified mario checks

### PR DESCRIPTION
**Pull request by crossroads1112**
Wanted to get input on this. It seemed wrong to have two identical `checks.py` files for more/less comfortable mario especially when they had the same parent package. This PR takes advantage of the `__init__.py` that they share to greatly reduce the amount of redundancy.

### CODE REVIEW

File: checks/cs50/2017/x/mario/mario.py [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+import sys
+
+sys.path.append(os.getcwd())
+from check50 import *
+
+class Mario(Checks):
+    @check()
+    def exists(self):
+        """mario.c exists."""
+        super(Mario, self).exists("mario.c")
```
Comment (dmalan) [inline checks/cs50/2017/x/mario/mario.py]: Did we decide this is the optimal Python 2/3 approach? I worry this is an API detail that a TF will always have to look up.

File: checks/cs50/2017/x/mario/mario.py [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+
+    @check("exists")
+    def compiles(self):
+        """mario.c compiles."""
+        self.spawn("clang -o mario mario.c -lcs50").exit(0)
+
+    @check("compiles")
+    def test_reject_negative(self):
+        """rejects a height of -1"""
+        self.spawn("./mario").stdin("-1").reject().kill()
```
Comment (dmalan) [inline checks/cs50/2017/x/mario/mario.py]: Is `kill` necessary, or could our post-check teardown just kill and still-open `Child`?

File: checks/cs50/2017/x/mario/mario.py [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+def check_pyramid(output, correct):
+    if output == correct:
+        return
+
+    err = Error((output, correct))
+    output = output.split("\n")
+    correct = correct.split("\n")
+
+    # check for trailing whitespace
+    if len(output) == len(correct) and all(ol.rstrip() == cl for ol, cl in zip(output, correct)):
```
Comment (dmalan) [inline checks/cs50/2017/x/mario/mario.py]: Let's definitely explain complexity like this in more of a comment for future TFs.

File: checks/cs50/2017/x/mario/more/checks.py [hunk 4]
```diff
@@ -1,79 +1,5 @@
...
-    err = Error((output, correct))
-    output = output.split("\n")
-    correct = correct.split("\n")
-
-    # check for trailing whitespace
-    if len(output) == len(correct) and all(ol.rstrip() == cl for ol, cl in zip(output, correct)):
-        err.helpers = "Did you add too much trailing whitespace to the end of your pyramid?"
-    raise err
+class MarioMore(Mario):
+    pass
```
Comment (dmalan) [inline checks/cs50/2017/x/mario/more/checks.py]: Should this now be in a `mario/more/more.py` file?

File: checks/cs50/2017/x/mario/mario.py (hunk 1 above)
Comment (crossroads1112) [inline checks/cs50/2017/x/mario/mario.py]: We didn't decide on what the best route was so I did it this way to resemble the other checks. Personally, I think `Checks.exists` is just as clear as `super().exists` when we have single inheritance (that's what StackOverflow seems to suggest anyway).

File: checks/cs50/2017/x/mario/more/checks.py (hunk 4 above)
Comment (crossroads1112) [inline checks/cs50/2017/x/mario/more/checks.py]: Per our discussion in slack, this approach enables certain slugs which are not intended. In any case, I wanted to do big changes like this all at once in a single commit after the PRs were merged so there aren't a bunch of unnecessary commits laying around.

File: checks/cs50/2017/x/mario/mario.py (hunk 2 above)
Comment (crossroads1112) [inline checks/cs50/2017/x/mario/mario.py]: 566590c

**Pull request closed without merging**